librz/core/cbin.c

```diff
@@ -738,11 +738,16 @@ RZ_API void rz_core_analysis_type_init(RzCore *core) {
 }
 
 RZ_API void rz_core_analysis_cc_init(RzCore *core) {
-	const char *dir_prefix = rz_config_get (core->config, "dir.prefix");
 	const char *analysis_arch = rz_config_get (core->config, "analysis.arch");
-	int bits = core->analysis->bits;
 	Sdb *cc = core->analysis->sdb_cc;
+	if (!strcmp (analysis_arch, "null")) {
+		sdb_reset (cc);
```

Comment:
why this? I don't see similar exit code in the other paths of this function.

Reply:
the null analysis plugin won't have a calling convention ever so might as well exit early ... or are you saying i don't need to `sdb_reset()`?

Reply:
if i'm not mistaken , `sdb_reset()` clears the cc sdb, basically saying that there's no calling conventions defined, which sounds reasonable to me

Reply:
The latter. Why do you need to call `sdb_reset`?

Reply:
Well `rz_core_analysis_cc_init()` does clear the cc sdb before doing anything with it:
https://github.com/rizinorg/rizin/blob/6601ae97596be4979675c8c358ee240aec0ebbbd/librz/core/cbin.c#L756-L757
so that tells me that i cannot assume that the cc sdb is empty when entering the function.

And since there are no calling conventions defined for the null analysis plugin, i should ensure that the sdb is empty before exiting.

Reply:
Mmmm ok then!